executor/src/witgen/affine_expression.rs

```diff
@@ -576,6 +574,34 @@ mod test {
         input.iter().map(|x| (*x).into()).enumerate().collect()
     }
 
+    #[test]
+    pub fn test_affine_assign() {
+        let mut a = AffineExpression::<_, GoldilocksField> {
+            coefficients: convert(vec![2, 3]),
+            offset: 3.into(),
+            clean: true,
+        };
+        a.assign(0, 3.into());
+        assert_eq!(
+            a,
+            AffineExpression {
+                coefficients: convert(vec![0, 3]),
+                offset: 9.into(),
+                clean: false
+            },
+        );
+
+        // Now, the expression is 3b + 9. It should be able to solve for b
+        // such that 3b + 9 = 0.
+        let updates = a.solve().unwrap();
+        for (col_id, update) in updates.constraints {
+            if let Constraint::Assignment(value) = update {
+                a.assign(col_id, value);
+            }
+        }
```

Comment:
Maybe something like
```suggestion
        assert_eq!(updates.constraints, [(0, Constraint::Assignment(-3))]);
        a.assign(0, -3);
```